Sources/Logging/Logging.swift

```diff
@@ -803,10 +919,69 @@ extension Logger {
+    /// using `LoggingSystem.bootstrap` is not appropriate for this particular logger.
+    ///
+    /// - parameters:
+    ///     - label: An identifier for the creator of a `Logger`.
+    ///     - metadataProvider: The custom metadata provider this logger should invoke,
+    ///                         instead of the system wide bootstrapped one, when a log statement is about to be emitted.
+    public init(label: String, metadataProvider: MetadataProvider) {
+        self = Logger(label: label, factory: { label in
+            LoggingSystem.factory(label, metadataProvider)
+        })
+    }
+
+    /// Construct a `Logger` given a `label` identifying the creator of the `Logger` or a non-standard ``Logger/metadataProvider`.
+    ///
+    /// The `label` should identify the creator of the `Logger`. This can be an application, a sub-system, or even
+    /// a datatype.
+    ///
+    /// This initializer provides an escape hatch in case the global default logging backend implementation (set up
+    /// using `LoggingSystem.bootstrap` is not appropriate for this particular logger.
+    ///
+    /// - parameters:
+    ///     - label: An identifier for the creator of a `Logger`.
+    ///     - factoryWithMetadataProvider: A closure creating non-standard `LogHandler`s.
+    #if swift(>=5.5) && canImport(_Concurrency) // we could instead typealias the function type, but it was requested that we avoid this for better developer experience
+    public init(label: String, metadataProvider: @escaping @Sendable() -> Metadata) {
+        self = Logger(label: label, factory: { label in
+            LoggingSystem.factory(label, .init(metadataProvider))
+        })
+    }
+
+    #else
+    public init(label: String, metadataProvider: @escaping () -> Metadata) {
+        self = Logger(label: label, factory: { label in
+            LoggingSystem.factory(label, .init(metadataProvider))
+        })
+    }
+    #endif
+
+    /// Construct a `Logger` given a `label` identifying the creator of the `Logger` or a non-standard `LogHandler`.
+    ///
+    /// The `label` should identify the creator of the `Logger`. This can be an application, a sub-system, or even
+    /// a datatype.
+    ///
+    /// This initializer provides an escape hatch in case the global default logging backend implementation (set up
+    /// using `LoggingSystem.bootstrap` is not appropriate for this particular logger.
+    ///
+    /// - parameters:
+    ///     - label: An identifier for the creator of a `Logger`.
+    ///     - factoryWithMetadataProvider: A closure creating non-standard `LogHandler`s.
+    public init(label: String, factoryWithMetadataProvider: (String, MetadataProvider?) -> LogHandler) {
+        self = Logger(label: label, factoryWithMetadataProvider(label, LoggingSystem.metadataProvider))
+    }
 }
 
 extension Logger.Level {
-    internal var naturalIntegralValue: Int {
```

Comment:
Seems unintentional

Reply:
Oh gosh don't tell me those slipped back in... we need to clean up how we do formatting, this is getting insane with the different versions of the tool in different projects... :|